src/NHapi.Base/Model/AbstractType.cs

```diff
@@ -25,75 +25,71 @@
+    ///
+    /// </summary>
+    /// <author>  Bryan Tripp.
+    /// </author>
+    public class AbstractType : IType
+    {
+        private ExtraComponents extra;
+        private IMessage message;
+        private string description;
 
-		/// <summary>
-		/// Extra components
-		/// </summary>
-		public virtual ExtraComponents ExtraComponents
-		{
-			get { return extra; }
-		}
+        /// <summary> Creates a new instance of AbstractType.</summary>
+        /// <param name="message">message to which this type belongs.
+        /// </param>
+        public AbstractType(IMessage message)
+            : this(message, null)
+        {
+        }
 
-		/// <returns> the message to which this Type belongs
-		/// </returns>
-		public virtual IMessage Message
-		{
-			get { return message; }
-		}
+        /// <summary> Creates a new instance of AbstractType.</summary>
+        /// <param name="message">message to which this type belongs.
+        /// <param name="description">The type description</param>
+        /// </param>
+        public AbstractType(IMessage message, string description)
+        {
+            extra = new ExtraComponents(message);
+            this.description = description;
+            this.message = message;
+        }
 
-		/// <summary>
-		/// Return the description of the type
-		/// </summary>
-		public virtual string Description
-		{
-			get { return description; }
-		}
+        /// <summary>
+        /// Returns the name of the type (used in XML encoding and profile checking).
+        /// </summary>
+        public virtual string TypeName
+        {
+            get
+            {
+                string longClassName = GetType().FullName;
+                return longClassName.Substring(longClassName.LastIndexOf('.') + 1);
+            }
+        }
 
+        /// <summary>
+        /// Extra components.
+        /// </summary>
+        public virtual ExtraComponents ExtraComponents
+        {
+            get { return extra; }
+        }
 
-		/// <summary> Creates a new instance of AbstractType</summary>
-		/// <param name="message">message to which this type belongs 
-		/// </param>
-		public AbstractType(IMessage message)
-			: this(message, null)
-		{
-		}
+        /// <returns> the message to which this Type belongs.
+        /// </returns>
+        public virtual IMessage Message
+        {
+            get { return message; }
+        }
 
-		/// <summary> Creates a new instance of AbstractType</summary>
-		/// <param name="message">message to which this type belongs 
-		/// <param name="description">The type description</param>
-		/// </param>
-		public AbstractType(IMessage message, string description)
-		{
-			extra = new ExtraComponents(message);
-			this.description = description;
-			this.message = message;
-		}
-	}
+        /// <summary>
+        /// Return the description of the type.
+        /// </summary>
+        public virtual string Description

```

Comment:
Can we get rid of the private field and just use the property?

Reply:
done